bot/seasons/valentines/lovecalculator.py

```diff
@@ -1,70 +1,94 @@
 
-with open(Path("bot", "resources", "valentines", "love_matches.json"), "r") as file:
+with Path('bot', 'resources', 'valentines', 'love_matches.json').open() as file:
     LOVE_DATA = json.load(file)
-LOVE_LEVELS = [int(x) for x in LOVE_DATA]
+    LOVE_DATA = sorted((int(key), value) for key, value in LOVE_DATA.items())
 
 
 class LoveCalculator:
     """
     A cog for calculating the love between two people
     """
+
     def __init__(self, bot):
         self.bot = bot
 
     @commands.command(aliases=('love_calculator', 'love_calc'))
-    @commands.cooldown(rate=1, per=5.0, type=commands.BucketType.user)
-    async def love(self, ctx, name_one: discord.Member, name_two: discord.Member = None):
+    @commands.cooldown(rate=1, per=5, type=commands.BucketType.user)
+    async def love(self, ctx, who: Union[Member, str], whom: Union[Member, str] = None):
         """
-        Calculates the love between two given names
+        Tells you how much the two love each other.
+
+        This command accepts users or arbitrary strings as arguments.
+        Users are converted from:
+          - User ID
+          - Mention
+          - name#discrim
+          - name
+          - nickname
 
-        DO NOT SPAM @mentions! There are five ways to hand over a name to this command:
-        1. ID of the user
-        2. Name + Discriminator of the user (name#discrim) (1)
-        3. Username (1)
-        4. Nickname (1)
-        5. @mention
+        Any two arguments will always yield the same result, though the order of arguments matters:
+          Running .love joseph erlang will always yield the same result.
+          Running .love erlang joseph won't yield the same result as .love joseph erlang
 
-        Using method 1-4 is highly encouraged, as nobody likes unwanted pings + they'll count as spam.
-        Skipping the second name will lead to something awesome.
+        If you want to use multiple words for one argument, you must include quotes.
+          .love "Zes Vappa" "morning coffee"
 
-        *(1): If the name has any form of spacing, the name must be wrapped inside quotes. Example:
-                .love Ves Zappa Niko Laus -> Will not work
-                .love "Ves Zappa" "Niko Laus" -> Will work
+        If only one argument is provided, the subject will become one of the helpers at random.
         """
 
-        if name_two is None:
+        if whom is None:
             staff = ctx.guild.get_role(Roles.helpers).members
-            name_two = choice(staff)
-
-        love_meter = (name_one.id + name_two.id) % 101
-        love_idx = str(sorted(x for x in LOVE_LEVELS if x <= love_meter)[-1])
-        love_status = choice(LOVE_DATA[love_idx]["titles"])
-
+            whom = random.choice(staff)
+
+        # if inputs were members, make sure to show name#discrim
+        # also make sure to escape all discord markdown to prevent mischief
+        cleaner = clean_content(escape_markdown=True, use_nicknames=False)
```

Comment:
Assuming the intent is to eliminate spoiler tags, this doesn't clean `str` inputs.

![image](https://user-images.githubusercontent.com/5323929/52902000-9b513c00-31d8-11e9-81ad-99f9aae5b1a9.png)

Reply:
works fine for me, are you sure your discord.py isn't out of date

Reply:
Whoops